src/main/java/ttt/game/Board.java

```diff
@@ -1,33 +1,45 @@
 package ttt.game;
 
-import java.util.Arrays;
+import java.util.ArrayList;
 
 public class Board {
-    private Object[] spaces;
+    private String[] spaces;
 
     public Board() {
-        spaces = new Object[] {1, 2, 3, 4, 5, 6, 7, 8, 9};
+        spaces = new String[] {"1", "2", "3", "4", "5", "6", "7", "8", "9"};
     }
 
-    public Board(Object[] state) {
+    public Board(String[] state) {
         spaces = state;
     }
 
-    public Object[] getSpaces() {
+    public String[] getSpaces() {
         return spaces;
     }
 
-    public Object[] availableSpaces() {
-        return Arrays.stream(spaces).filter(space -> space instanceof Integer).toArray();
+    public ArrayList<Integer> availableSpaces() {
+        ArrayList<Integer> availableSpaces = new ArrayList<>();
+        int index = 0;
+
+        for(String space : spaces) {
+            try {
```

Comment:
Does this need to be a try? Are you ever going to be in a position where the value cannot be an integer?

Generally, in Java, we shouldn't use exceptions as the basis of determining normal program flow of execution. i.e. try not to use it to implement "if this is a number add its index to the list".

Reply:
I agree that this feels a bit overblown - I got to this as a means of determining whether a space was empty - in which case it should be a string representation of an integer - or taken - in which case it will be a string 'X' or 'O'.

I'll look again for a more elegant solution.